odxtools/parameters/createanyparameter.py

```diff
@@ -27,205 +23,33 @@
 def create_any_parameter_from_et(et_element: ElementTree.Element,
                                  doc_frags: List[OdxDocFragment]) \
                                  -> Parameter:
-    kwargs = dataclass_fields_asdict(NamedElement.from_et(et_element, doc_frags))
-    semantic = et_element.get("SEMANTIC")
-    byte_position_str = et_element.findtext("BYTE-POSITION")
-    byte_position = int(byte_position_str) if byte_position_str is not None else None
-    bit_position_str = et_element.findtext("BIT-POSITION")
-    bit_position = None
-    if bit_position_str is not None:
-        bit_position = int(bit_position_str)
     parameter_type = et_element.get(f"{xsi}type")
 
-    sdgs = create_sdgs_from_et(et_element.find("SDGS"), doc_frags)
-
     # Which attributes are set depends on the type of the parameter.
-    dop_ref = None
-    dop_snref = None
-    if parameter_type in ["VALUE", "PHYS-CONST", "SYSTEM", "LENGTH-KEY"]:
-        dop_ref = OdxLinkRef.from_et(et_element.find("DOP-REF"), doc_frags)
-        dop_snref = None
-        if (dop_snref_elem := et_element.find("DOP-SNREF")) is not None:
-            dop_snref = odxrequire(dop_snref_elem.get("SHORT-NAME"))
-
-        if dop_ref is None and dop_snref is None:
-            raise ValueError(
-                f"A parameter of type {parameter_type} must reference a DOP! {dop_ref}, {dop_snref}"
-            )
-
     if parameter_type == "VALUE":
```

Comment:
instead of multiple ifs, maybe use a lookup dict?

Reply:
other than this, LGTM

Reply:
that would make things harder to read and I doubt that the performance would be measurably better...